src/Compilers/CSharp/Test/Semantic/Diagnostics/DiagnosticAnalyzerTests.AllInOne.cs

```diff
@@ -86,7 +86,7 @@ public void AnalyzerDriverIsSafeAgainstAnalyzerExceptions()
         {
             var compilation = CreateCompilationWithMscorlib45(TestResource.AllInOneCSharpCode, parseOptions: TestOptions.Regular);
             ThrowingDiagnosticAnalyzer<SyntaxKind>.VerifyAnalyzerEngineIsSafeAgainstExceptions(analyzer =>
-                compilation.GetAnalyzerDiagnostics(new[] { analyzer }, null, CodeAnalysis.DiagnosticExtensions.AlwaysCatchAnalyzerExceptions), AnalyzerDriverHelper.DiagnosticId);
+                compilation.GetAnalyzerDiagnostics(new[] { analyzer }, null, CodeAnalysis.DiagnosticExtensions.AlwaysCatchAnalyzerExceptions));
```

Comment:
previously, we would have never passed around func like primary way to change behavior of same code.

either should hierarchy of a type

IDEAnalyzerDriver { override bool ContinueOnException }, CompilerAnalyzerDriver { override bool Continue ...}

or ExcentpionHandlerFactory.CreateCrashOnBuiltInAnalyzerExceptionHandler(), ExceptionHandlerFactory.CreateC.... which return something like IDiagnosticExceptionHandler.

rather than passing around func.